resources/metakey.py

```diff
@@ -311,27 +339,38 @@ def put(self, key, group_name):
             description: Attribute key permission definition
             content:
                 application/json:
-                  schema: MetakeyPermissionSchema
+                  schema: MetakeyPermissionSetRequestBodySchema
         responses:
             200:
                 description: When group permission has been successfully changed
+                content:
+                  application/json:
+                    schema: MetakeyDefinitionManageItemResponseSchema
             400:
                 description: When one of attribute permission fields is missing or incorrect.
             403:
                 description: When user doesn't have `managing_attributes` capability.
             404:
                 description: When attribute key or group doesn't exist
         """
-        schema = MetakeyPermissionSchema()
+        schema = MetakeyPermissionSetRequestArgsSchema()
+        args_obj = schema.load({
+            "key": key,
+            "group_name": group_name
+        })
+        if args_obj.errors:
+            return {"errors": args_obj.errors}, 400
+
+        schema = MetakeyPermissionSetRequestBodySchema()
         obj = schema.loads(request.get_data(as_text=True))
         if obj.errors:
             return {"errors": obj.errors}, 400
 
-        metakey_definition = db.session.query(MetakeyDefinition).filter(MetakeyDefinition.key == key)
+        metakey_definition = db.session.query(MetakeyDefinition).filter(MetakeyDefinition.key == key).first()
         if metakey_definition is None:
             raise NotFound("No such metakey")
 
-        group = db.session.query(Group).filter(Group.name == obj.data["group_name"]).first()
+        group = db.session.query(Group).filter(Group.name == group_name).first()
```

Comment:
Wouldn't reading it from `obj.data` be safer? In case the group is modified (or normalized) inside the schema?

Reply:
Nope, `obj.data` doesn't have that information anymore because it was redundant with arguments provided by path. Single argument should be passed only in one place.

Reply:
Yup, we should use `args_obj.data["group_name"]` here.

Reply:
Fixed